flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/TableEnvironment.scala

```diff
@@ -358,20 +358,27 @@ abstract class TableEnvironment(val config: TableConfig) {
     * Registers an [[AggregateFunction]] under a unique name. Replaces already existing
     * user-defined functions under this name.
     */
-  private[flink] def registerAggregateFunctionInternal[T: TypeInformation, ACC](
+  private[flink] def registerAggregateFunctionInternal[T: TypeInformation, ACC: TypeInformation](
       name: String, function: AggregateFunction[T, ACC]): Unit = {
     // check if class not Scala object
     checkNotSingleton(function.getClass)
     // check if class could be instantiated
     checkForInstantiation(function.getClass)
 
-    val typeInfo: TypeInformation[_] = implicitly[TypeInformation[T]]
+    val resultTypeInfo: TypeInformation[_] = implicitly[TypeInformation[T]]
+    val accTypeInfo: TypeInformation[_] = implicitly[TypeInformation[ACC]]
 
     // register in Table API
     functionCatalog.registerFunction(name, function.getClass)
 
     // register in SQL API
-    val sqlFunctions = createAggregateSqlFunction(name, function, typeInfo, typeFactory)
+    val sqlFunctions = createAggregateSqlFunction(
```

Comment:
Hi @fhueske , I think the ACC type extraction is similar to `T` type of `AggreateFunction[T, ACC]` and `T` type of `TableFunction[T]`. Currently, they are both extracted and add to the internal representation of UDAGGs and UDTFs. To keep the consistent, I followed the same way to add the acc type. If we decide to adopt the approach that wrapping the registered functions, it would be better to make the other type extractions follow this approach.

What do you think ?